### PR TITLE
Added nGraph as a public dependency

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Add nGraph as a public dependency for Inference Engine

### CODE REVIEW

File: inference-engine/include/cpp/ie_cnn_network.h [hunk 1]
```diff
@@ -20,12 +20,7 @@
 #include "ie_common.h"
 #include "ie_data.h"
 #include "ie_extension.h"
-
-namespace ngraph {
-
-class Function;
-
-}  // namespace ngraph
+#include <ngraph/ngraph.hpp>
```
Comment (ilya-lavrenov) [inline inference-engine/include/cpp/ie_cnn_network.h:23]: can we include only `ngraph/function.hpp`

File: inference-engine/include/ie_extension.h [hunk 2]
```diff
@@ -14,6 +14,7 @@
 #include <string>
 #include <vector>
 
+#include <ngraph/ngraph.hpp>
```
Comment (ilya-lavrenov) [inline inference-engine/include/ie_extension.h:17]: the same

File: inference-engine/src/inference_engine/CMakeLists.txt [hunk 3]
```diff
@@ -216,7 +217,7 @@ set_target_properties(${TARGET_NAME} ${TARGET_NAME}_obj ${TARGET_NAME}_s
 
 # Export for build tree
 
-export(TARGETS ${TARGET_NAME} NAMESPACE IE::
+export(TARGETS ${NGRAPH_LIBRARIES} ${TARGET_NAME} NAMESPACE IE::
```
Comment (ilya-lavrenov) [inline inference-engine/src/inference_engine/CMakeLists.txt:220]: do we need it? Currently `InferenceEngineConfig.cmake` already includes `find_package(ngraph REQUIRED)`

File: ngraph/core/include/ngraph/op/asin.hpp [hunk 4]
```diff
@@ -34,7 +34,11 @@ namespace ngraph
 
                 virtual std::shared_ptr<Node>
                     clone_with_new_inputs(const OutputVector& new_args) const override;
-                bool visit_attributes(AttributeVisitor& visitor) override { return true; }
+                bool visit_attributes(AttributeVisitor& visitor) override
+                {
```
Comment (ilya-lavrenov) [inline ngraph/core/include/ngraph/op/asin.hpp:38]: can we just remove `visitor` as an argument?

File: inference-engine/src/inference_engine/CMakeLists.txt (hunk 3 above)
Comment (ilyachur) [reply inference-engine/src/inference_engine/CMakeLists.txt:220]: Looks like we don't need.

File: inference-engine/include/ie_parameter.hpp [hunk 5]
```diff
@@ -20,7 +20,7 @@
 #include <vector>
 
 #include "ie_blob.h"
-#include <ngraph/ngraph.hpp>
+#include <ngraph/variant.hpp>
```
Comment (ilya-lavrenov) [inline inference-engine/include/ie_parameter.hpp:23]: is it needed? We don't have `ngraph::Variant` integration here

Comment (ilyachur) [reply inference-engine/include/ie_parameter.hpp]: Good catch. Looks like variant support was removed

File: inference-engine/src/inference_engine/CMakeLists.txt (hunk 3 above)
Comment (ilyachur) [reply inference-engine/src/inference_engine/CMakeLists.txt:220]: Reverted, because some targets cannot find IE::ngraph.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**